components/filesystem/src/std_filesystem.cpp

```diff
@@ -0,0 +1,154 @@
+/* Copyright (c) 2024, Thomas Atkinson
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "std_filesystem.hpp"
+
+#include <core/util/logging.hpp>
+
+#include <filesystem>
+#include <fstream>
+
+namespace vkb
+{
+namespace filesystem
+{
+FileStat StdFileSystem::stat_file(const Path &path)
+{
+	std::error_code ec;
+
+	auto fs_stat = std::filesystem::status(path, ec);
+	if (ec)
+	{
+		return FileStat{
+		    false,
+		    false,
+		    0,
+		};
+	}
+
+	auto size = std::filesystem::file_size(path, ec);
+	if (ec)
+	{
+		size = 0;
+	}
+
+	return FileStat{
+	    fs_stat.type() == std::filesystem::file_type::regular,
+	    fs_stat.type() == std::filesystem::file_type::directory,
+	    size,
+	};
+}
+
+bool StdFileSystem::is_file(const Path &path)
+{
+	auto stat = stat_file(path);
+	return stat.is_file;
+}
+
+bool StdFileSystem::is_directory(const Path &path)
+{
+	auto stat = stat_file(path);
+	return stat.is_directory;
+}
+
+bool StdFileSystem::exists(const Path &path)
+{
+	auto stat = stat_file(path);
+	return stat.is_file || stat.is_directory;
+}
```

Comment:
All of these end up calculating the size which isn't used.  It seems like they might be better implmented using `std::filesystem::status` directly.